x/supplier/client/cli/query_claim_test.go

```diff
@@ -84,64 +95,151 @@ func createClaim(
 	}
 }
 
+func getSessionId(
+	t *testing.T,
+	net *network.Network,
+	appAddr string,
+	supplierAddr string,
+	sessionStartHeight int64,
+) (string, error) {
+	t.Helper()
+	ctx := context.TODO()
+
+	sessionQueryClient := sessiontypes.NewQueryClient(net.Validators[0].ClientCtx)
+	res, err := sessionQueryClient.GetSession(ctx, &sessiontypes.QueryGetSessionRequest{
+		ApplicationAddress: appAddr,
+		Service:            &sharedtypes.Service{Id: "svc1"}, // hardcoded for simplicity
+		BlockHeight:        sessionStartHeight,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	var found bool
+	for _, supplier := range res.GetSession().GetSuppliers() {
+		if supplier.GetAddress() == supplierAddr {
+			found = true
+			break
+		}
+	}
+	require.Truef(t, found, "supplier address %s not found in session", supplierAddr)
+
+	return res.Session.SessionId, nil
+}
+
+// TODO_CONSIDERATION: perhaps this (and/or other similar helpers) can be refactored
+// into something more generic and moved into a shared testutil package.
 func networkWithClaimObjects(
 	t *testing.T,
-	numSessions int,
-	numClaimsPerSession int,
+	sessionCount int,
+	supplierCount int,
+	appCount int,
+	// TODO_THIS_COMMIT: hook up to genesis state generation...
+	serviceCount int,
 ) (net *network.Network, claims []types.Claim) {
 	t.Helper()
 
-	// Prepare the network
+	// Initialize a network config.
 	cfg := network.DefaultConfig()
+
+	// Construct an in-memory keyring so that it can be populated and used prior
+	// to network start.
+	kr := keyring.NewInMemory(cfg.Codec)
+	// Populate the in-memmory keyring with as many pre-generated accounts as
```

Comment:
`as we expect to need for the test`

Can `20` be a function of all the counts provided to the function?